### DISCUSSION TITLE
BPA Provide format string for string measures?

### DISCUSSION THREAD

**Discussion by ricky7uio** [TabularEditor/TabularEditor3#1291] [🙋♀️TE3 Q&A] [answered]
I am trying to clean my models enforcing BPAs. As a starting point I am using the official list provided by Microsoft. 
I came across several violations triggered by this rule:
![image](https://github.com/TabularEditor/TabularEditor3/assets/44656050/4222969c-f0b9-42ee-b6d8-f7940d614c8f)

The rule totally make sense in the case the measure is a date or number. But what about string measures? Does it really make sense (any examples)? 

Because if not, I was thinking on refining it a bit to exclude string measures. 

Thanks for sharing your opinions/views.

**Comment by mlonsk** [comment, marked as answer]
Hi @ricky7uio 
I would agree that for measures returning strings the rule should not apply.

I double checked in Power BI how a Text measure is formatted (1 in picture bellow) and it also does not get a format string (2), but has a Data Type of 'String/Text'

![image](https://github.com/TabularEditor/TabularEditor3/assets/11939114/67a3846d-5d14-4905-905d-8cea2c1cb76f)

So my conclusion is that the rule could very well exclude String measures.

**Comment by ricky7uio** [comment]
Thanks for checking and for your reply! It's very much appreciated :)
